### ISSUE TITLE
Issue with oneOf and allOf aggregators for the same objects (bidirectional)

### ISSUE THREAD

**Issue by wtrocki** [oasdiff/oasdiff#163]
**Describe the bug**

kin-openapi fails with circular dependencies when number of objects in allOf or oneOf aggregator are bigger than 3. 
Example schema: https://gist.github.com/wtrocki/2e272375386ddecff69d6228c89d2a05

This happens because oasdiff doesn't change default (very conservative) value in kin-openapi library:

`CircularReferenceCounter`

**To Reproduce**
Steps to reproduce the behavior:}
0. Download openapi file https://gist.github.com/wtrocki/2e272375386ddecff69d6228c89d2a05
1. Run `oasdiff   -base openapi.yaml -revision openapi.yaml`
2. see error Failed to load base spec from "circularapi.yaml" with kin-openapi bug found: circular schema reference not handled - #/components/schemas/OnlineArchiveScheduleView -> #/components/schemas/DefaultScheduleView -> #/components/schemas/OnlineArchiveScheduleView -> #/components/schemas/DailyScheduleView -> #/components/schemas/OnlineArchiveScheduleView -> #/components/schemas/WeeklyScheduleView -> #/components/schemas/OnlineArchiveScheduleView

**Expected behavior**
OpenAPI is parsed

**Comment by wtrocki** [comment]
Ideally the value should be controlled by new flag.

**Comment by reuvenharrison** [comment]
Hi @wtrocki and thanks for your contribution.

I have one question please:
Why did you set the default value to zero?
Wouldn't it be better to align with the default in kin-openapi of three (3):
```
var CircularReferenceError = "kin-openapi bug found: circular schema reference not handled"
var CircularReferenceCounter = 3
```

**Comment by wtrocki** [comment]
I have changed that value to be 5. After reviewing and debugging parser code I do not think that increasing that value to something like 20 etc. would not have any performance impact. This is only to prevent out of memory/stack recurrence issues. So changing value to 20 would only increase amount of jumps in cases where someone have 20 references (which is not uncommon)

**Comment by wtrocki** [comment]
IMHO 3 is way to low as it is quite common for base class to have 3 subclasses etc.
I was thinking of hardcoding some large number without flag there but that would be hard to determine as we never know how complex inheritance patterns oas-diff users might have.

**Comment by reuvenharrison** [comment]
Fixed by #164
